src/main/java/com/google/devtools/build/lib/analysis/BuildView.java

```diff
@@ -433,31 +434,37 @@ public AnalysisResult update(
               topLevelTargetsWithConfigsResult,
               /*includeExecutionPhase=*/ true);
     } else {
-      TopLevelConstraintSemantics topLevelConstraintSemantics =
-          new TopLevelConstraintSemantics(
-              (RuleContextConstraintSemantics) ruleClassProvider.getConstraintSemantics(),
-              skyframeExecutor.getPackageManager(),
-              input -> skyframeExecutor.getConfiguration(eventHandler, input),
-              eventHandler);
-
-      PlatformRestrictionsResult platformRestrictions =
-          topLevelConstraintSemantics.checkPlatformRestrictions(
-              skyframeAnalysisResult.getConfiguredTargets(), explicitTargetPatterns, keepGoing);
-
-      if (!platformRestrictions.targetsWithErrors().isEmpty()) {
-        // If there are any errored targets (e.g. incompatible targets that are explicitly specified
-        // on the command line), remove them from the list of targets to be built.
-        skyframeAnalysisResult =
-            skyframeAnalysisResult.withAdditionalErroredTargets(
-                platformRestrictions.targetsWithErrors());
-      }
+      Set<ConfiguredTarget> targetsToSkip;
+
+      if (reportIncompatibleTargets) {
+        TopLevelConstraintSemantics topLevelConstraintSemantics =
+            new TopLevelConstraintSemantics(
+                (RuleContextConstraintSemantics) ruleClassProvider.getConstraintSemantics(),
+                skyframeExecutor.getPackageManager(),
+                input -> skyframeExecutor.getConfiguration(eventHandler, input),
+                eventHandler);
+
+        PlatformRestrictionsResult platformRestrictions =
+            topLevelConstraintSemantics.checkPlatformRestrictions(
+                skyframeAnalysisResult.getConfiguredTargets(), explicitTargetPatterns, keepGoing);
+
+        if (!platformRestrictions.targetsWithErrors().isEmpty()) {
+          // If there are any errored targets (e.g. incompatible targets that are explicitly
+          // specified on the command line), remove them from the list of targets to be built.
+          skyframeAnalysisResult =
+              skyframeAnalysisResult.withAdditionalErroredTargets(
+                  platformRestrictions.targetsWithErrors());
+        }
 
-      Set<ConfiguredTarget> targetsToSkip =
-          Sets.union(
-                  topLevelConstraintSemantics.checkTargetEnvironmentRestrictions(
-                      skyframeAnalysisResult.getConfiguredTargets()),
-                  platformRestrictions.targetsToSkip())
-              .immutableCopy();
+        targetsToSkip =
+            Sets.union(
+                    topLevelConstraintSemantics.checkTargetEnvironmentRestrictions(
+                        skyframeAnalysisResult.getConfiguredTargets()),
+                    platformRestrictions.targetsToSkip())
+                .immutableCopy();
+      } else {
+        targetsToSkip = ImmutableSet.of();
```

Comment:
Nit: Set this back on line 437 (where targetsToSkip is declared) and remove the `else` clause.

Reply:
Done.